File: backend/app/Models/Commande.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

class Commande extends Model
{
    public $timestamps = false;

    protected $fillable = [
        'numero_commande',
        'acheteur_id',
        'stock_id',
        'produit',
        'quantite',
        'prix',
        'date_commande',
        'statut',
    ];

    protected static function booted(): void
    {
        static::creating(function (Commande $commande) {
            if (empty($commande->numero_commande)) {
                $commande->numero_commande = 'CMD-' . now()->format('Ymd') . '-' . strtoupper(Str::random(4));
            }
        });
    }

    public function acheteur()
    {
        return $this->belongsTo(AcheteurGros::class, 'acheteur_id');
    }

    public function stock()
    {
        return $this->belongsTo(Stock::class, 'stock_id');
    }

    public function livraison()
    {
        return $this->hasOne(Livraison::class, 'commande_id');
    }
}

File: mobile-app/lib/models/commande.dart
class Commande {
  final int id;
  final String numeroCommande;
  final String produit;
  final double quantite;
  final double? prix;
  final String? dateCommande;
  final String statut;
  final bool alerteLivraison;
  final Map<String, dynamic>? livraison;
  final Map<String, dynamic>? stock;

  const Commande({
    required this.id,
    required this.numeroCommande,
    required this.produit,
    required this.quantite,
    this.prix,
    this.dateCommande,
    required this.statut,
    this.alerteLivraison = false,
    this.livraison,
    this.stock,
  });

  factory Commande.fromJson(Map<String, dynamic> j) => Commande(
        id: j['id'],
        numeroCommande: j['numero_commande'] ?? '',
        produit: j['produit'],
        quantite: (j['quantite'] as num).toDouble(),
        prix: (j['prix'] as num?)?.toDouble(),
        dateCommande: j['date_commande'],
        statut: j['statut'],
        alerteLivraison: j['alerte_livraison'] == true,
        livraison: j['livraison'],
        stock: j['stock'],
      );
}
